uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/my-leave/components/my-leave-calendar.js

```diff
@@ -9,21 +11,352 @@ define([
     templateUrl: ['settings', function (settings) {
       return settings.pathTpl + 'components/my-leave-calendar.html';
     }],
-    controllerAs: 'calendar',
-    controller: ['$log', controller]
+    controllerAs: 'ctrl',
+    controller: ['$log', '$q', 'OptionGroup', 'AbsencePeriod', 'AbsenceType',
+      'Calendar', 'PublicHoliday', 'LeaveRequest', controller]
   });
 
 
-  function controller($log) {
+  function controller($log, $q, OptionGroup, AbsencePeriod, AbsenceType, Calendar, PublicHoliday, LeaveRequest) {
     $log.debug('Component: my-leave-calendar');
 
-    init.call(this);
+    var vm = Object.create(this),
+      dayTypes = [],
+      leaveRequests = [],
+      publicHolidays = [],
+      leaveRequestStatuses = [];
+
+    vm.absencePeriods = [];
+    vm.absenceTypes = [];
+    vm.calendar = {};
+    vm.loading = false;
+    vm.months = ['January', 'February', 'March', 'April', 'May', 'June',
+      'July', 'August', 'September', 'October', 'November', 'December'];
+    //Select current month as default
+    vm.selectedMonths = [vm.months[new Date().getMonth()]];
+    vm.selectedPeriod = null;
+
+    /**
+     * Decides whether sent date is a public holiday
+     *
+     * @param  {string} date
+     * @return {boolean}
+     */
+    vm.isPublicHoliday = function (date) {
+      return !!publicHolidays[new Date(date).getTime()];
+    };
+
+    /**
+     * Labels the given period according to whether it's current or not
+     *
+     * @param  {AbsencePeriodInstance} period
+     * @return {string}
+     */
+    vm.labelPeriod = function (period) {
+      return period.current ? 'Current Period (' + period.title + ')' : period.title;
+    };
+
+    /**
+     * Returns day name of the sent date(Monday, Tuesday etc.)
+     *
+     * @param  {string} date
+     * @return {string}
+     */
+    vm.getDayName = function (date) {
+      return new Date(date).toString().substr(0,3);
+    };
+
+    /**
+     * Returns the calendar information for a specific month
+     *
+     * @param  {int} month
+     * @return {array}
+     */
+    vm.getMonthData = function (month) {
+      if (vm.calendar.days) {
```

Comment:
The check is necessary to because before `WorkPattern`data is loaded the `days` object is empty, so as the function is used in the `view`, we are getting errors.